File: 3-longest-substring-without-repeating-characters/longest-substring-without-repeating-characters.py
class Solution(object):
    def lengthOfLongestSubstring(self, s):
        """
        :type s: str
        :rtype: int
        """
        if s=="":
            return 0
        i=1
        sub_str=""
        sub_str+=s[0]
        length=1
        while i<len(s):
            x=sub_str.find(s[i])
            if x!=-1:
                sub_str=sub_str[x+1:]
                sub_str+=s[i]
            else:
                sub_str+=s[i]
            i+=1 
            length=max(length,len(sub_str))
        return length 

File: 3-longest-substring-without-repeating-characters/longest-substring-without-repeating-characters.cpp
class Solution {
public:
    int lengthOfLongestSubstring(string s) {
        int ans=1,j=0;
        if(s=="") return 0;
        unordered_map<char,int> freq;
        for(int i=0;i<s.size();i++){
            freq[s[i]]++;
            while(j<s.size() && freq[s[i]]>1){
                freq[s[j]]--;
                j++;
            }
            ans=max(ans,i-j+1);
        }
        return ans;
    }
};